scRNA-seq-advanced/01-reading_filtering_CellRanger.Rmd

```diff
@@ -0,0 +1,484 @@
+But before we do that, we might want to quickly plot to see what would be filtered out with a given cutoff.
+The default is to exclude cells that have a posterior probability of 0.75 or greater of being compromised.
+
+
+```{r miQC plotFiltering}
+miQC::plotFiltering(filtered_sce, miqc_model, posterior_cutoff = 0.75)
+```
+
+In this case seems to correspond to a mitochondrial percentage of about 12.5%, but note that this will not always be constant: the cutoff point can vary for different numbers of unique genes.
+
+At this point, we can perform the actual filtering using the `filterCells()` function, giving us a further filtered SCE object.
+
+```{r miQC filtercells}
+qcfiltered_sce <- miQC::filterCells(filtered_sce, model = miqc_model)
+```
+
+### One more filter: unique genes
+
+While the miQC filtering is pretty good, you may have noticed that it still left some cells that had very low numbers of unique genes. 
+While these cells may not be compromised, the information from them is also not likely to be useful, so we will filter those as well. 
+For now, we will use a cutoff of 200 unique genes as a minimum. 
+
+```{r unique cutoff}
+qcfiltered_sce <- qcfiltered_sce[, which(qcfiltered_sce$detected >= 200)]
+```
+
+
+
+## Normalization and Dimension Reduction
+
+Now that we have done our filtering, we can set analyzing the expression counts for the remaining cells. 
+
+The next step at this point is to convert the raw counts into a measure that accounts for differences in sequencing depth between cells, and to convert the distribution of expression values from the skewed distribution we expect to see in raw counts to one that is more normally distributed.
+
+We will do this using functions from the `scran` and `scuttle` packages. 
+The procedure we will use here is derived from [Orchestrating Single-Cell Analysis with Bioconductor](http://bioconductor.org/books/3.16/OSCA.basic/normalization.html#normalization-by-deconvolution).
+The idea is to first account for the fact that different cell types may have different expression patterns that could affect the normalization factors we would calculated if we were to use the raw sequencing depth. To account for that, we first do a rough clustering of cells, then use that clustering to calculate the scaling factor (sum factor) for each cell within the cluster.
+Finally, we apply the scaling factor and calculate the log-scaled expression values.
+
+```{r normalization}
+# Perform rough clustering
+qclust <- scran::quickCluster(qcfiltered_sce)
+
+# use clusters to compute scaling factors and add to SCE object
+qcfiltered_sce <- scran::computeSumFactors(qcfiltered_sce, clusters = qclust)
+
+# perform normalization using scaling factors
+# and save as a new SCE object
+normalized_sce <- scuttle::logNormCounts(qcfiltered_sce)
+```
+
+This creates a new "assay" in the `normalized_sce` object, `logcounts`, which contains the normalized count values for each cell and gene.
```

Comment:
"Sadly these are not actually the log of the counts, but this term is used for historical reasons." We can include or not, up to you!